src/actinia_core/core/utils.py

```diff
@@ -49,11 +50,22 @@ def os_path_normpath(path_parts):
         raises a SecurityError if the path is not as expected
     """
     path = os.path.normpath(os.path.join(*path_parts))
-    if not path.startswith(path_parts[0]):
+    if not path.startswith(path_parts[0]) or not path_is_in_allowed_paths(path):
         raise SecurityError(f"Used path '{path}' is insecure")
     return path
 
 
+def path_is_in_allowed_paths(path):
+    # TODO we should double check whether they are all needed, better would be one
+    # single location
+    allowed_starts = [
+        global_config.GRASS_USER_DATABASE,
+        global_config.GRASS_DATABASE,
+        global_config.GRASS_RESOURCE_DIR
+    ]
+    return any(path.startswith(allowed) for allowed in allowed_starts)
```

Comment:
Theoretically the `startswith` could be unsafe here again, if  `path="GRASS_RESOURCE_DIR/../../something_else`.

Reply:
`os.path.normpath("GRASS_RESOURCE_DIR/../../something_else")` leads to `../something_else`, so the normpath is needed that this function `path_is_in_allowed_paths` is working. Maybe you can mention that path should be a path which was corrected with `os.path.normpath`.

Reply:
But if `os.path.normpath("GRASS_RESOURCE_DIR/../../something_else")` leads to `../something_else` the function would return false, as it does not start with any of the allowed prefixes. I can of course add a note